core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -101,24 +107,69 @@ public static BlockchainSourceImpl getInstance() {
 	}
 
 	@Override
-	public void createAccount() throws BlockchainException {
-		createKinAccountIfNeeded();
+	public void loadAccount(String kinUserId) throws BlockchainException {
+		migrateToMultipleUsers(kinUserId);
+		createOrLoadAccount(kinUserId);
 		initBalance();
 	}
 
-	private void createKinAccountIfNeeded() throws BlockchainException {
-		if (kinClient.hasAccount()) {
+	/**
+	 * Support backward compatibility,
+	 * load the old account and migrate to current multiple users implementation.
+	 */
+	private void migrateToMultipleUsers(String kinUserId) {
+		final boolean isMigrated = local.getIsMigrated();
+		if (!isMigrated && kinClient.hasAccount()) {
+			final int accountIndex = local.getAccountIndex();
+			KinAccount account;
+			if(accountIndex == NOT_EXIST) {
+				account = kinClient.getAccount(0);
+				Logger.log(new Log().withTag(TAG).put("migrateToMultipleUsers accountIndex == NOT_EXIST, kinUserId", kinUserId));
+			} else {
+				Logger.log(new Log().withTag(TAG).put("migrateToMultipleUsers accountIndex", accountIndex).put("kinUserId", kinUserId));
+				account = kinClient.getAccount(accountIndex);
+				local.deleteAccountIndexKey();
+			}
+			local.setDidMigrate();
```

Comment:
check first  'if (!isMigrated)' and call `local.setDidMigrate();` otherwise first time user will not migrate for the first time but in the second time isMigrate will still be false but this time hasAccount will be true. In an edge case where User A log in for the first time logout and login back as user B they will share account.